src/Illuminate/Support/Arr.php

```diff
@@ -67,6 +82,38 @@ public static function collapse($array)
     }
 
     /**
+     * Creates an array by using one array for keys and another for its values.
+     *
+     * @param  array|\ArrayAccess  $items
+     * @param  bool  $includeKeyDeprivedValues
+     * @return array
+     */
+    public static function combine($keys, $values, $includeKeyDeprivedValues = true)
+    {
+        $combined = [];
+
+        if ($keys instanceof Collection) {
+            $keys = $keys->all();
+        }
+
+        if ($values instanceof Collection) {
+            $values = $values->all();
+        }
+
+        for ($i = 0; $i < count($values); $i++) {
+            if (isset($keys[$i]) && static::canBeString($keys[$i]) && (string) $keys[$i] !== '') {
+                static::set($combined, (string) $keys[$i], $values[$i]);
+            } elseif ($includeKeyDeprivedValues) {
+                array_push($combined, $values[$i]);
+            } else {
+                break;
```

Comment:
When combining keys `['foo']` and values `['bar', 'baz', 'qux', 'norf']` the loop would run 4 times, yet the resulting array will be `['foo' => 'bar']`, which could be done in just 2 iterations.